### PR TITLE
MULE-17091: Object streams not automatically closed if <non-repeatabl…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…e-iterable> option is selected

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/streaming/bytes/factory/NullCursorStreamProviderFactory.java [hunk 1]
```diff
@@ -38,7 +39,7 @@ public NullCursorStreamProviderFactory(ByteBufferManager bufferManager, Streamin
    */
   @Override
   protected Object resolve(InputStream inputStream, EventContext creatorRootEventContext) {
-    streamingManager.manage(inputStream, creatorRootEventContext);
+    streamingManager.manage((Closeable) inputStream, creatorRootEventContext);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/streaming/bytes/factory/NullCursorStreamProviderFactory.java]: could `inputStream` not be a closeable here?

File: core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java [hunk 2]
```diff
@@ -149,11 +150,16 @@ public CursorProvider manage(CursorProvider provider, EventContext creatorEventC
    */
   @Override
   public void manage(InputStream stream, EventContext creatorEventContext) {
-    if (stream instanceof Cursor) {
+    manage((Closeable) stream, creatorEventContext);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: could `stream` not be a closeable here?

File: core/src/main/java/org/mule/runtime/core/internal/streaming/bytes/factory/NullCursorStreamProviderFactory.java (hunk 1 above)
Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/internal/streaming/bytes/factory/NullCursorStreamProviderFactory.java]: nope. InputStream class implements Closeable

File: core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java (hunk 2 above)
Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: same answer

File: core-tests/src/test/java/org/mule/runtime/core/internal/util/queue/LocalTxQueueTransactionTestCase.java [hunk 3]
```diff
@@ -86,7 +86,7 @@ private LocalTxQueueTransactionContext createLocalTxContext(LocalTxQueueTransact
     });
     return new LocalTxQueueTransactionContext(txLog, provider, lock);
   }
-  
+
```
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/internal/util/queue/LocalTxQueueTransactionTestCase.java]: missing rebase?

**Pull request merged**